packages/ui/src/ui/UIFactory.tsx

```diff
@@ -355,6 +356,10 @@ export interface UIFactory {
 
     renderRolesLink(params: {cluster: string; login: string; className?: string}): React.ReactNode;
 
+    getCustomQueryTabName(): string;
+
+    renderCustomQueryTab(params: {query: QueryItem}): React.ReactNode;
```

Comment:
Do we really need two separate methods? I think it is better to leave only one factory method, like:
```ts
getCustomQueryResultTab(): undefined | {title: string, renderContent() => React.ReactNode};
```